euphonic/spectra.py

```diff
@@ -336,7 +337,7 @@ def __init__(self, x_data: Quantity, y_data: Quantity,
         self.x_tick_labels = x_tick_labels
         self.metadata = {} if metadata is None else metadata
 
-    def __add__(self, other: S1D):
+    def __add__(self, other: T) -> T:
```

Comment:
Looking at the implementation, I think this one is actually guaranteed to return Spectrum1D even if inherited? It doesn't look like it would be based on the type of `other` directly.

Reply:
Yes, maybe that is misleading. Would `(self: T, other: T) -> T` be better? We can't use `Spectrum1D` directly because Py 3.6 doesn't support forward references

Reply:
`from_spectra` explicitly requires that the input must all be Spectrum1D. Can we not forward-reference a type hint using quotes? So it becomes
```python
def __add__(self, other: 'Spectrum1D') -> 'Spectrum1D'
```